### PR TITLE
Deduplicate the logic behind `risk_category`

### LINKED ISSUE #324: DRY the call to `case_when()`
Relates to #212 

Here is some code we need to DRY`*`: The call to `case_when()` in these three places could likely be extracted in a single function that can then be reused in for all indicators. If there is any difference in the logic (e.g. `>` in one bue `>=` in another one) I would suspect a bug.

* [pstr](https://github.com/2DegreesInvesting/tiltIndicator/blob/0812197e3184d06104cdaa0186e9b022fa40ab41/R/pstr.R#L77)
* [istr](https://github.com/2DegreesInvesting/tiltIndicator/blob/0812197e3184d06104cdaa0186e9b022fa40ab41/R/istr.R#L64)
* [xctr](https://github.com/2DegreesInvesting/tiltIndicator/blob/0812197e3184d06104cdaa0186e9b022fa40ab41/R/xctr_at_product_level.R#L109)

Before refactoring consider adapting [these tests](https://github.com/2DegreesInvesting/tiltIndicator/blob/0812197e3184d06104cdaa0186e9b022fa40ab41/tests/testthat/test-pstr.R#L74). 

--

`*` DRY means Dont Repeat Yourself and it's commonly used as a verb to mean "deduplicate".

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #324 

This PR deduplicates the logic behind `risk_category`. All indicators so far do it in the same way, and now use a low level `categorize_risk()` that thinly wraps `case_when()`. 

We could deduplicate further but the challenge is now that `xctr*()` currently works with wide data (values spread across multiple columns) while `xstr*()` works with long data (values gathered in the same column). Using dplyr, the approach by `xstr*()` is simpler, so I would follow up with a PR the makes `xctr*()` work also with long data.

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR title and description.
- [x] Assign a reviewer.

EXCEPTIONS

- [ ] Slide here any item that you intentionally choose to not do.
- [ ] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).

This is purely refactoring so the existing tests are enough.

### CODE REVIEW

File: R/istr.R [hunk 1]
```diff
@@ -38,13 +39,17 @@ istr <- function(companies, scenarios, mapper) {
 
 #' @rdname istr
 #' @export
-istr_at_product_level <- function(companies, scenarios, mapper) {
+istr_at_product_level <- function(companies,
+                                  scenarios,
+                                  mapper,
+                                  low_threshold = 30,
+                                  high_threshold = 70) {
```
Comment (maurolepore) [inline R/istr.R:46]: See https://github.com/2DegreesInvesting/tiltIndicator/issues/332

File: R/istr.R [hunk 2]
```diff
@@ -38,13 +39,17 @@ istr <- function(companies, scenarios, mapper) {
 
 #' @rdname istr
 #' @export
-istr_at_product_level <- function(companies, scenarios, mapper) {
+istr_at_product_level <- function(companies,
+                                  scenarios,
+                                  mapper,
+                                  low_threshold = 30,
```
Comment (lindadelacombaz) [inline R/istr.R:45]: I am not familiar with istr so I do not know - but just wanted to flag that here the thresholds are between 1-100 not like pstr. if this is correct my comment can be ignored ! :)

File: R/pstr.R [hunk 3]
```diff
@@ -71,16 +71,13 @@ pstr_add_reductions <- function(companies, scenarios) {
...
-        reductions <= low_threshold ~ "low",
-        reductions > low_threshold & reductions <= high_threshold ~ "medium",
-        reductions > high_threshold ~ "high",
-        TRUE ~ "no_sector",
-      )
-    )
+xstr_categorize_risk <- function(data,
+                                 low_threshold,
+                                 high_threshold,
+                                 .default = "no_sector") {
```
[the remark this answers is not in the document]
Comment (maurolepore) [reply R/pstr.R:77]: Yeah, thanks for noting that connection. For the record, the related open issue is https://github.com/2DegreesInvesting/tiltIndicator/issues/300. That needs  a lot more brain power than this refactoring, so yeah, I think it should be tackled separately.

[the remark this answers is not in the document]
Comment (maurolepore) [reply R/pstr.R:77]: I note that https://github.com/2DegreesInvesting/tiltIndicator/issues/300#issuecomment-1542778879 suggests the existing assertion would block `NA`s in `reductions`, and further suggests that means the `.default` of `"no_sector"` would never run. However I've seen `"no_sector"` [here](https://github.com/2DegreesInvesting/tiltIndicator/issues/315#issuecomment-1546355895) so it's not so straight forward.

**Pull request merged**